src/main/java/com/mcneilio/orcmaker/BasicEventDriver2.java

```diff
@@ -0,0 +1,144 @@
+package com.mcneilio.orcmaker;
+
+import com.mcneilio.orcmaker.columns.ColumnInitializer;
+import com.mcneilio.orcmaker.orcer.Orcer;
+import com.mcneilio.orcmaker.storage.StorageDriver;
+import com.timgroup.statsd.StatsDClient;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.ColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.OrcFile;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.Writer;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Properties;
+import java.util.UUID;
+
+
+public class BasicEventDriver2 implements EventDriver {
+  Orcer[] orcers;
+
+  String[] colNames;
+  VectorizedRowBatch batch;
+  TypeDescription schema;
+  HashMap<String, ColumnVector> columns;
+  String eventName;
+  String eventDate;
+  String fileName;
+  String statsdEnv;
+  Configuration conf = new Configuration();
+  Writer writer = null;
+  StatsDClient statsd;
+  StorageDriver storageDriver;
+  ColumnVector[] columnVectors;
+
+
+  public BasicEventDriver2(Properties config, TypeDescription typeDescription, StorageDriver storageDriver, StatsDClient statsd) {
```

Comment:
I was trying not to expose the internal classes `StorageDriver` and `StatsDClient` explicitly. Plan was to eventually make them package-only and accessed purely through property configuration.

Reply:
was thinking more dependency injection so it would be easier to mock in tests, and more single responsibility.

Reply:
I like the dependency injection idea. I added support to the `BasicEventDriver`. Though I omitted `StatsDClient` for now, I like the direction you're headed with a `MetricStrategy` interface that allows the same flexibility as the `StorageDriver`